CONTRIBUTING.md

```diff
@@ -0,0 +1,116 @@
+
+All source code is formatted using [prettier](https://prettier.io).
+When code is modified and committed, all staged files will be automatically
+formatted using pre-commit git hooks (with help from the
+[lint-staged](https://github.com/okonet/lint-staged) and
+[husky](https://github.com/typicode/husky) libraries). The benefit of using a
+code formatter like prettier is that it removes the topic of code style from the conversation
+when reviewing pull requests, thereby speeding up the review process.
+
+You may also use the prettier npm script (e.g. `npm run prettier` or `yarn prettier` or `jlpm prettier`) to format the entire code base. We recommend
+installing a prettier
+extension for your code editor and configuring it to format your code with
+a keyboard shortcut or automatically on save.
+
+## Setting Up a Development Environment
+
+### Creating the environment using conda
+
+Building JupyterLab Metadata Service from its GitHub source code requires Node.js and JupyterLab.
+
+If you use `conda`, you can get it with:
+
+```bash
+conda create -n jupyterlab-metadata-service -c conda-forge --override-channels nodejs jupyterlab
+
+# and activate your new environment
+conda activate jupyterlab-metadata-service
+
+```
+
+## Build and install the extension for development
+
+Run the following commands to install the initial project dependencies and install it in the JupyterLab environment.
+
+```bash
+
+jlpm install
+jupyter labextension install . --no-build
+
+```
+
+After the install completes, open a second terminal. Run these commands to activate the `jupyterlab-metadata-service`
+environment and to start a JupyterLab instance in watch mode so that it will keep up with our changes as we make them.
+
+
+```bash
+
+conda activate jupyterlab-ext
+jupyter lab --watch
+
+```
+
+### Build and Run the Tests
+
+```bash
+jlpm run build:test
+jlpm test
+```
+
+## [Writing Documentation](#writing-documenation)
+
+Documentation is written in Markdown and reStructuredText. In particular, the documentation on our Read the Docs page is written in reStructuredText. To ensure that the Read the Docs page builds, you'll need to install the documentation dependencies with `conda`. These dependencies are located in `docs/environment.yml`. You can install the dependencies for building the documentation by creating a new conda environment:
+
+```bash
+conda env create -f docs/environment.yml
+```
+
+Alternatively, you can install the documentation dependencies in an existing environment using the following command:
+
+```bash
+conda env update -n <ENVIRONMENT> -f docs/environment.yml
+```
+
+The Developer Documentation includes a [guide](http://jupyterlab.readthedocs.io/en/latest/developer/documentation.html) to writing documentation including writing style, naming conventions, keyboard shortcuts, and screenshots.
+
+To test the docs run:
+
+```
+py.test --check-links -k .md . || py.test --check-links -k .md --lf .
```

Comment:
I think we need this dep added to `docs/environment.yml` for this command to work: https://github.com/minrk/pytest-check-links

Reply:
you're right. thanks!

Reply:
I needed to add a link to the github project because the pip version is very old and I needed a updated related to config.warn (that was removed on pytest)